eskip/eskip.go

```diff
@@ -320,3 +324,119 @@ func GenerateIfNeeded(existingId string) string {
 	id = routeIdRx.ReplaceAllString(id, "x")
 	return "route" + id
 }
+
+func marshalJsonPredicates(r *Route) []*Predicate {
+	rjf := make([]*Predicate, 0)
```

Comment:
some nitpicking would be to set the capacity: `rjf := make([]*Predicate, 0, len(r.Predicates))`. Not significant in this case.